### ISSUE TITLE
aggregate $match Error: Type 'ObjectId' is not assignable to type 'Expression'.

### ISSUE THREAD

**Issue by sm-hejazi** [Automattic/mongoose#11952]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.4.0

### Node.js version

16.14.0

### MongoDB server version

5.0.9

### Description

update to latest version and get error on aggregate
```javascript
  await Model.aggregate([
                  {
                      $match: { _id: new mongoose.Types.ObjectId('stringObjecId') },
                  },
  ]);
```
get Error `Type 'ObjectId' is not assignable to type 'Expression'.`
how to solve this?

### Steps to Reproduce

```javascript
  await Model.aggregate([
                  {
                      $match: { _id: new mongoose.Types.ObjectId('stringObjecId') },
                  },
  ]);
```

**Comment by andreialecu** [comment]
I opened #11956, which should fix this.

**Comment by jyeros** [comment]
I am having a similar issue, I am trying `$ne: ['$$indexNotNull', null]`, but types doesn't allow null as a second parameter. A fix could be add null as second parameter with an union type. But, I think null is a literal expression in mongo.

**Comment by vkarpov15** [comment]
@jyeros this should be fixed in 6.4.3 with https://github.com/Automattic/mongoose/commit/9f3f8ebb951a37cdb43934c38e6501782e35fa2f .
